openadmet_toolkit/cheminf/data_curation.py

```diff
@@ -0,0 +1,240 @@
+            )
+            clean_deduped["action_type"] = clean_deduped["action_type"].apply(
+                lambda x: x.lower() if isinstance(x, str) else x
+            )
+        else:
+            clean_deduped["action_type"] = "substrate"
+        clean_deduped["dataset"] = "ChEMBL_curated"
+        clean_deduped["active"] = True
+        if save_as is not None:
+            clean_deduped.to_csv(save_as, index=False)
+        return clean_deduped
+
+    @classmethod
+    def get_more_than_N_compounds(self, combined):
+        more_than_N_compounds = combined[combined["molecule_count"] > N]
+        more_than_N_compounds.INCHIKEY = more_than_N_compounds.INCHIKEY.astype(str)
+        return more_than_N_compounds["Assay ChEMBL ID"].nunique()
+
+    @classmethod
+    def get_num_assays_per_compound(self, more_than_N_compounds):
+        num_assays_per_compound_df = (
+            more_than_N_compounds.groupby(["INCHIKEY"])["Assay ChEMBL ID"]
+            .size()
+            .reset_index(name="assay_count")
+        )
+        return num_assays_per_compound_df.set_index("INCHIKEY")
+
+    @classmethod
+    def get_num_compounds_per_assay(self, better_units):
+        num_compounds_per_assay = better_units.groupby("Assay ChEMBL ID")[
+            "Molecule ChEMBL ID"
+        ].nunique()
+        num_compounds_per_assay_df = pd.DataFrame(num_compounds_per_assay)
+        num_compounds_per_assay_df.rename(
+            columns={"Molecule ChEMBL ID": "molecule_count"}, inplace=True
+        )
+        return better_units.join(num_compounds_per_assay_df, on="Assay ChEMBL ID")
+
+    @classmethod
+    def aggregate_activity(self, combined_2):
+        compound_grouped_mean = combined_2.groupby("INCHIKEY")["pChEMBL Value"].mean()
+        return compound_grouped_mean.reset_index()
+
+    @classmethod
+    def get_num_assays_per_compound(self, more_than_N_compounds):
+        num_assays_per_compound_df = (
+            more_than_N_compounds.groupby(["INCHIKEY"])["Assay ChEMBL ID"]
+            .size()
+            .reset_index(name="assay_count")
+        )
+        return num_assays_per_compound_df.set_index("INCHIKEY")
+
+
+class PubChemProcessing(CSVProcessing):
+    """
+    Class to handle processing data from a csv downloaded
+    from PubChem
+
+    """
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.pubChemData = self.read_csv(self.csv_path)
+        self.keep_cols = [
+            "CANONICAL_SMILES",
+            "INCHIKEY",
+            "PUBCHEM_ACTIVITY_OUTCOME",
+            "PUBCHEM_CID",
+        ]
+        self.delete_metadata_rows()
+        self.pubChemData = self.pubChemData.dropna(subset="PUBCHEM_CID")
+        self.pubChemData["PUBCHEM_SID"] = self.pubChemData["PUBCHEM_SID"].astype(int)
+        self.pubChemData["PUBCHEM_CID"] = self.pubChemData["PUBCHEM_CID"].astype(int)
+        self.standardize_smiles_and_convert()
+        self.pubChemData.dropna(subset="INCHIKEY")
```

Comment:
This is good, much closer to getting it done in one integrated call, probably just better outside of `__init__` (ie in something like `process`.